pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -195,3 +196,40 @@ func validateMachine(machine machinev1.Machine, providerSpec v1beta1.GCPMachineP
 	// Default values can also be set here
 	return nil
 }
+
+// Returns true if machine exists.
+func (r *Reconciler) instanceExists() (bool, error) {
+	if err := validateMachine(*r.machine, *r.providerSpec); err != nil {
+		return false, fmt.Errorf("failed validating machine provider spec: %v", err)
+	}
+	zone := r.providerSpec.Zone
+	// Need to verify that our project/zone exists before checking machine, as
+	// invalid project/zone produces same 404 error as no machine.
+	if err := r.validateZone(); err != nil {
```

Comment:
should this be invoked by `validateMachine()`

Reply:
@enxebre I don't think so.  Also, that would result in an external google API call for pretty much every function we implement, which is probably not good.

We should call it before create though (or, just call instanceExists in-line in create).